src/server/db_setup.ts

```diff
@@ -182,3 +182,75 @@ export async function insertGameResults(
     // TODO: Could return latest DB results here
     return { status: 200, results: [] };
 }
+
+// Checks the existence of an ID in the database
+// Returns list of errors, one for each duplicate ID
+export async function checkGameIdExists(
+    gameId: string[]
+): Promise<ErrorObject[]> {
+    // Check if the game already exists in the table
+    let params: string[] = [];
+    for (let i = 1; i <= gameId.length; i++) {
+        params.push('$' + i);
+    }
+
+    let queryText: string =
+        'SELECT * FROM game_results WHERE game_label IN (' +
+        params.join(',') +
+        ')';
+    const res = await pool.query(queryText, gameId);
+    let allErrors: ErrorObject[] = [];
+
+    for (let row of res.rows) {
+        const dupError: ErrorObject = {
+            status: 'error',
+            error: 'Duplicate ID entered, no data uploaded: '.concat(
+                row.game_label
+            )
+        };
+        allErrors.push(dupError);
+    }
+
+    return allErrors;
+}
+
+//to test data upload
+//when page is refreshed, submitted data shows up in raw results table
+// This is the new function that can handle multiple insertions
+export async function insertGameResults(
+    allReq: GameResultsForm[]
+): Promise<GameResultsFormResult> {
+    let result;
+    let allErrors: ErrorObject[] = [];
+
+    // Check all game IDs being inserted
+    const allIds: string[] = [];
```

Comment:
If you want a fancy one-liner of this (just sharing to point out nice features of JS):
`const allIds = allIds.map(req => req.gameId);`